Code/autopkglib/AppPkgCreator.py

```diff
@@ -100,7 +106,10 @@ def package_app(self, app_path):
         infoplist = self.read_info_plist(app_path)
         if not self.env.get("version"):
             try:
-                self.env["version"] = infoplist["CFBundleShortVersionString"]
+                self.env["version"] = (
+                    infoplist.get(self.env.get("version_key"))
+                    or infoplist["CFBundleShortVersionString"]
```

Comment:
You can combine these lines:
```
infoplist.get(self.env.get("version_key"), infoplist["CFBundleShortVersionString"])
```

However, this still has a safety risk in that it's _possible_ none of those keys are defined. I would consider something more explicit that informs the admin of a serious problem, which combines your solution and mine:
```
infoplist.get(self.env.get("version_key"), infoplist["CFBundleShortVersionString"]) or "0.0.0.0 (FIX ME!)"
```

If you want to be extra safe, consider adding an additional check at the end that if you don't have a valid result (either `None` or the `"0.0.0.0"` idea), throw a ProcessorError, because at that point you'll have a bad version you can't do anything about, and proceeding with it is likely dangerous.

Reply:
Sure, I agree with the code flow, but my point is that it doesn't particularly give the operator a lot of useful information. You and I are people familiar with Python would likely recognize that a key error for this indicates something important is missing, but anyone not fluent in the underlying code here would be confused, as they'd see a traceback without any context or information on what they can do about it. 

I'd like to try and make AutoPkg's failure states more meaningful for a wider audience, so ideally you'd capture this error and present a useful indication - i.e. (paraphrasing here) "go yell at the developer because this is really broken and there's nothing we can do to fix it", or "there's no way to get version information from this because it's malformed".

I don't _particularly_ care that much which particular way you solve it, I just want to make the output very explicit.

Reply:
I agree with that philosophy. The exception traps I've seen are very broad. Let me put something together and add it to my PR.